portal/app/[locale]/staking-dashboard/_hooks/useIncreaseAmount.ts

```diff
@@ -0,0 +1,174 @@
+    token.chainId,
+  )
+
+  const updateNativeBalanceAfterFees = useUpdateNativeBalanceAfterReceipt(
+    token.chainId,
+  )
+
+  const { hemiWalletClient } = useHemiWalletClient()
+
+  const { allowanceQueryKey } = useNeedsApproval({
+    address: token.address,
+    amount,
+    spender: veHemiAddress,
+  })
+
+  return useMutation({
+    mutationFn: function runIncreaseAmount() {
+      if (!address) {
+        throw new Error('No account connected')
+      }
+      const { emitter, promise } = increaseAmount({
+        account: address,
+        additionalAmount: amount,
+        approvalAdditionalAmount: amount,
+        tokenId: BigInt(tokenId),
+        walletClient: hemiWalletClient!,
+      })
+
+      emitter.on('user-signed-approve', function (approvalTxHash) {
+        updateStakingDashboardOperation({
+          approvalTxHash,
+          status: StakingDashboardStatus.APPROVAL_TX_PENDING,
+          transactionHash: undefined,
+        })
+      })
+      emitter.on('approve-transaction-reverted', function (receipt) {
+        updateStakingDashboardOperation({
+          status: StakingDashboardStatus.APPROVAL_TX_FAILED,
+        })
+
+        updateNativeBalanceAfterFees(receipt)
+
+        track?.('staking dashboard - approve increase amount reverted')
+      })
+      emitter.on('approve-transaction-succeeded', function (receipt) {
+        updateStakingDashboardOperation({
+          status: StakingDashboardStatus.APPROVAL_TX_COMPLETED,
+        })
+
+        updateNativeBalanceAfterFees(receipt)
+        queryClient.invalidateQueries({ queryKey: allowanceQueryKey })
+      })
+      emitter.on('user-signed-increase-amount', function (transactionHash) {
+        updateStakingDashboardOperation({
+          stakingPosition: { amount, tokenId },
+          status: StakingDashboardStatus.STAKE_TX_PENDING,
+          transactionHash,
+        })
+
+        track?.('staking dashboard - signed increase amount')
+      })
+      emitter.on('user-signing-increase-amount-error', function () {
+        updateStakingDashboardOperation({
+          status: StakingDashboardStatus.STAKE_TX_FAILED,
+        })
+
+        track?.('staking dashboard - signing increase amount error')
+      })
+      emitter.on('increase-amount-transaction-succeeded', function (receipt) {
+        updateStakingDashboardOperation({
+          status: StakingDashboardStatus.STAKE_TX_CONFIRMED,
+        })
+
+        queryClient.setQueryData(
+          stakingPositionQueryKey,
+          (old: StakingPosition[] | undefined = []) =>
+            old.map(position =>
+              position.tokenId === tokenId
+                ? { ...position, amount: position.amount + amount }
+                : position,
+            ),
+        )
+
+        // fees
+        updateNativeBalanceAfterFees(receipt)
+        // HEMI balance
+        queryClient.setQueryData(
+          hemiBalanceQueryKey,
+          (old: bigint) => old + amount,
+        )
```

Comment:
I think here is the other way around. You're adding $HEMI to your locked position, thus reducing your HEMI balance in your wallet

```suggestion
        queryClient.setQueryData(
          hemiBalanceQueryKey,
          (old: bigint) => old - amount,
        )
```